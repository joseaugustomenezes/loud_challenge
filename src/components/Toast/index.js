import React, { useState, useEffect } from "react";
import BootstrapToast from "react-bootstrap/Toast";

const Toast = ({ status, message }) => {
  const [show, setShow] = useState(false);

  useEffect(() => {
    status && setShow(true);
  }, [status])

  return (
    <div style={{ position: "absolute", top: 10, right: 20 }}>
      <BootstrapToast onClose={() => setShow(false)} show={show || false} delay={3000} autohide>
        <BootstrapToast.Header>
          <strong className="mr-auto">Error {status}</strong>
        </BootstrapToast.Header>
        <BootstrapToast.Body>{message}</BootstrapToast.Body>
      </BootstrapToast>
    </div>
  );
};

export default Toast;
