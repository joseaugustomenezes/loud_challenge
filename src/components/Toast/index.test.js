import React from 'react';
import { shallow, mount } from 'enzyme';

import Toast from './index';

describe('<Toast />', () => {
  describe('no props', () => {
    const container = shallow(<Toast />);

    it('no props should render without errors', () => {
      expect(container.html()).toMatchSnapshot();
    });
  })

  describe('with props', () => {
    const initialProps = {
      status: 500,
      message: 'test message'
    };
    const container = mount(<Toast {...initialProps} />);
    const clickfn = jest.fn();

    it('should render without errors', () => {
      expect(container.find('strong').prop('children')).toEqual(["Error ", 500]);
      expect(container.find('div.toast-body').prop('children')).toEqual('test message');
    });
  });
});