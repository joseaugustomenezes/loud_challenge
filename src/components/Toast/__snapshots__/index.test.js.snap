// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> no props no props should render without errors 1`] = `"<div style=\\"position:absolute;top:10px;right:20px\\"></div>"`;
